### PR TITLE
IS-15302 Add support for Purchase Transactions endpoint

### PR DESCRIPTION
**Pull request by parhambrz**

### CODE REVIEW

File: templates/purchaseinvoices.json [hunk 1]
```diff
@@ -0,0 +1,153 @@
+[
+  {
+    "_id": "{{@ system @}}-{{@ datatype @}}-all",
+    "namespaced_identifiers": false,
+    "source": {
+      "operation": "list",
+      "properties": {
+        "endpoint": "purchase_invoices",
+        "primary_key": "id"
```
Comment (ellis-igneri) [inline templates/purchaseinvoices.json]: What is the primary_key property under properties for? I don't see it referenced in the list operation?

File: templates/purchaseinvoicesrelease.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
+{
+  "_id": "{{@ system @}}-{{@ datatype @}}-share",
```
Comment (ellis-igneri) [inline templates/purchaseinvoicesrelease.json]: This is probably the wrong way to do this, lets talk.

File: templates/purchaseinvoices.json (hunk 1 above)
Comment (parhambrz) [reply templates/purchaseinvoices.json]: I see. I think it's not needed there. I duplicated other pipes and edited them. Think I may need to clean up some other data types as well.

File: templates/purchaseinvoices.json [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+              ]
+            ]
+          ],
+          "invoice-lines-rule": [
+            [
+              "copy",
+              "*",
+              [
+                "list",
+                "ledger_account"
```
Comment (ellis-igneri) [inline templates/purchaseinvoices.json]: You will almost certainly need products and tax_rates for invoice lines post-mapping so can you add the necessary rewrite here?

Comment (parhambrz) [reply templates/purchaseinvoices.json]: These are not required by the api, but yes I can add them.

**Review by ellis-igneri — APPROVED** [review]

**Review by ellis-igneri — APPROVED** [review]

**Pull request merged**